File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int n=nums.size();
        int lastseen=-1;
        int i=0,j=0,res=0;
        while(j<n){
            if(nums[j]==0){
                i=lastseen+1;
                lastseen=j;
            }
            res=max(res,j-i);
            j++;
        }
        return res;
    }
};

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int n=nums.length;
        int zeros=0;
        int l=0,r=0,max=0;
        while(r<n){
            if(nums[r]==0){
                zeros++;
            }
            if(zeros>1){
                if(nums[l]==0){
                    zeros--;
                }
                l++;
            }
            max=Math.max(max,r-l+1);
            r++;
        }
        return max-1;
    }
}